### ISSUE TITLE
How strong is Leela Zero?

### ISSUE THREAD

**Issue by pw31** [leela-zero/leela-zero#916]
The progress figure at http://zero.sjeng.org shows an ELO-scale with respect to random player=0. Is there any possibility to add a kyu/dan/pro scale, at least approximately? I guess this is what most people would like to know, and I have no clue how to interprete ELO-over-random. Maybe take one version and let it auto-play on KGS for a while to determine rank? That would be fantastic!

**Comment by petgo3** [comment]
see this topic #898

**Comment by pw31** [comment]
Hm, I read that topic before I submitted this one, but failed to spot a proper answer to my question.  Is it correct that "RoyalLeela" on KGS (7d) is running the latest Lela Zero version? So that 8500 ELO would mean about 7dan KGS?

**Comment by wctgit** [comment]
I'm guessing RoyalLeela is not Leela Zero, but just the original Leela. On OGS, that's how it's named also. The 'zero' version is called RoyalZero.

**Comment by emc2314** [comment]
This [Google Doc](https://docs.google.com/spreadsheets/d/1QhYWYQvXBhI5eHJm45fvf97CaV2rRmgrATwVgBFGKoU/) might help.

**Comment by pw31** [comment]
Thank you, emc2314, that google-doc is very informative.  Am I correct summarizing that Leela Zero achieved about 2900 "Bayes ELO" (or 8500 "ELO") after 100 days of training, which puts it currently somewhere in between of Pachi (2d KGS -- 2700 Bayes ELO) and Zen 5 (6d KGS -- 3300 Bayes ELO)?  The "Bayes ELO" rating is anchored at -3451 for random play and has spacing 148 ELO / kyu grade. However, the "ELO" scaling on http://zero.sjeng.org is different, is anchored at 0 for random play, and has a different spacing. If we assume a linear relation, we have roughly

    BAYES ELO ~ -3451 + 0.74*ELO

so if Leela Zero wants to catch up with Alpha Zero (Bayes ELO~4500) it has to achieve about 10600 ELO on http://zero.sjeng.org?

Really ?!?

**Comment by WhiteHalmos** [comment]
@pw31 This "Bayes Elo" you are referring to is for a specific server, namely CGOS. On that server, ratings are anchored to Gnu Go, which is set at 1800. In AlphaGo papers Gnu Go is anchored somewhere around 350-450 Elo rating. So the -3451 random Elo from them are not directly comparable to CGOS Elo.

**Comment by pw31** [comment]
@WhiteHalmos  Thanks, I see, it's probably a bit more complicated than I thought.  But if I am mistaken, and I guess I was, what please is the bottom line of that Google Doc?  Thank you!

**Comment by d7urban** [comment]
As a perhaps interesting data point, I ran LeelaZero with the 8ee49-net against Leela 11, each with 30 secs/move. LeelaZero won 3 out of 5, with one of the losses a misunderstood ladder and the other a giant semeai. I hope our training will close those two weaknesses soon.

**Comment by wctgit** [comment]
Recent _match_ games on http://zero.sjeng.org often come down to either ladders (less frequently recently) or large semeais (i.e. large dying dragons), so it seems very likely to me that recent training is indeed 'working on' these kinds of weaknesses. 

Recent _training_ games rarely show ladders, probably due to using the same search tree for both players, but also often come down to large semeais (or several medium sized ones simultaneously), so I think that lends some credence to the idea that training will is currently working on it and will probably get steadily better at it as we go.

**Comment by pw31** [comment]
@petgo3 : Aha, now I understand your comment. There is a player "petgo3" on KGS which is used as bot-account to run Leela Zero. It is currently ranked 4d on KGS, quickly increasing ... impressive!

**Comment by WhiteHalmos** [comment]
@pw31 5d now. 😉

**Comment by pw31** [comment]
@WhiteHalmos: yes, petgo3 on KGS just promoted to 5d. And RoyalZeroSlow on OGS to 8d!

**Comment by wctgit** [comment]
@petgo3 Could you include the name of the neural network being used in the comments at the beginning of the game?

For those wondering, here's the link: http://www.gokgs.com/gameArchives.jsp?user=petgo3

**Comment by kityanhem** [comment]
maybe this can help you: https://www.reddit.com/r/cbaduk/comments/7jx8tv/the_ranking_of_lzero/

**Comment by d7urban** [comment]
I just finished a 30 game match between Leela11 and LZ b3b00. All matches both sides got 30 minutes on the clock. Leela11 won 16 and LZ won 14. I think we're very close to that milestone, where we overtake Leela11!

**Comment by jkiliani** [comment]
If this was with b3b00, I think there's an excellent chance the first 10 block net is going to defeat Leela 11.

**Comment by pw31** [comment]
@d7urban : could you repeat that exercise for the new f1383525 network?  Would be very interesting to evaluate the progress of the new 10-block network

**Comment by d7urban** [comment]
Will do

**Comment by barrtgt** [comment]
Ran a 50 game match earlier with b3b against camwagner's strong supervised net with 1.6k playouts, b3b took 34 games.

**Comment by pw31** [comment]
LeelaZ_1080_5b90 just defeated best Zen version (Zen-15.7-3c1g, ELO 3924 on CGOS) for the very first time, see http://www.yss-aya.com/cgos/19x19/SGF/2018/03/06/391974.sgf  :-)

**Comment by d7urban** [comment]
@pw31 Match is halfway done, LZ with net 5b90 has won 12 out 15 games so far

**Comment by fergusyuen** [comment]
hi, there is a popular online go game server called 野狐围棋 in china, its developed by Tencent company. there are also many chinese, japanese and korean pro players like Ke Jie, playing on that server. are there any possiblilies that someone use leela zero on that server so that we can know how strong is leela zero?

**Comment by roy7** [comment]
alphaleela account on Fox is Leela Zero.

**Comment by alreadydone** [comment]
Hi there, Fox server (野狐围棋) has formerly been the major testground of Leela Zero in China, however after some unpleasant episodes (including Tencent banning our QQ group with 1100+ members), we've switched to yikeweiqi (弈客围棋), on which there are also strong professional players all with real names. Recently there are matches to raise funds to rent machines to generate self-play games.
There may still be some active accounts like alphaleela that serve as official test accounts on Fox, but actually it's said all non-pro accounts on Fox 9d should be using Leela Zero by now.

**Comment by fergusyuen** [comment]
oops, i didnt notice that. so whenever the latest version of leela zero is released, they will test leelazero on those servers. am i right?

**Comment by alreadydone** [comment]
My understanding is that people are not so keen to test the latest official network these days because it's improving slowly. Instead they use LZ data to train or hybridize to get stronger networks. Recently they are able to reach Tianrang 5P (highest level) and consistently beat DeepZenGo on Tygem. The machines they use differ; it's [said](https://share.yikeweiqi.com/gonews/detail/id/16450?from=timeline) Titan V x 4 will be used in the fundraising tournament. To get info about how tests are going, look into the following QQ groups and ask people:
Leela Zero | 726658329  (1000+ members)
Leela Zero testing | 604275603
Hybrid weights | 555788099  (20x256)
Leela Master | 693862763  (20x128, trained with human games)
irrelevant: Lizzie developers | 638934756

On other servers, @roy7 is running the latest official network on online-go.com. It should also be possible for @gcp to replace the official Leela account on Tygem by LZ.

**Comment by fergusyuen** [comment]
thankyou very much for the information!

**Comment by ThomasCabaret** [comment]
I do some experiments on my personal computers, mainly between leela bestnet and leela11 with handicap stones.
Leela11: --noponder -t1 --nobook -g
LeelaZero: -w d04568336 --noponder -t1 -r1 -g
Leelazero seems to confidently win leela11 with H2.
I never had any H3 win until this morning (surprisingly on one of my personal computers where I do not use GPU). It's only "almost" a win because leelaz managed to actually win the game, but put itself in atari at the end of the yose... I did undo it's bad move, I connected the stones instead, then I let both programs conclude that it's an actual win of leelaz vs leela11 H3.
Here is the game, the small variation at the end is the self atari of leelaz.
http://eidogo.com/#FHMY5Iop
